app/cmd/volume.go

```diff
@@ -36,6 +36,27 @@ func ExpandCmd() cli.Command {
 	}
 }
 
+func UnmapMarkSnapChainRemovedCmd() cli.Command {
+	return cli.Command{
+		Name:      "unmap-mark-snap-chain-removed",
+		ShortName: "unmap-mark-snap",
+		Flags: []cli.Flag{
+			cli.BoolFlag{
+				Name: "enable",
+			},
+			cli.BoolFlag{
+				Name: "disable",
+			},
```

Comment:
After enabling the flag, will the following newly created snapshots be marked as `removed`?

Reply:
If there is an unmap request immediately after the snapshot creation, then the answer is YES.
I guess you are worried about the backup creation. Yes this may lead to the following backup creation fail.

Reply:
Actually, I don't feel comfortable with the separation of the sync agent server and the replica server now. Due to the newly introduced feature like unmap or prune, the removed snapshot is not immutable anymore hence the rebuilding flow should be aware of it and be protected by the replica lock. Otherwise, things may mess up...

Reply:
Added a TODO comment in `func (c *Controller) UnmapAt(length uint32, off int64) (int, error)`